cli/tests/unit/http_test.ts

```diff
@@ -2614,6 +2614,129 @@ Deno.test({
   },
 });
 
+async function httpServerWithErrorBody(
+  listener: Deno.Listener,
+  compression: boolean,
+): Promise<Deno.HttpConn> {
+  const conn = await listener.accept();
+  listener.close();
+  const httpConn = Deno.serveHttp(conn);
+  const e = await httpConn.nextRequest();
+  assert(e);
+  const { respondWith } = e;
```

Comment:
maybe you should assert something here about the request being H1 or H2

Reply:
Let me see if I can find a way to distinguish them. Unfortunately we do not expose the HTTP version on the `Request` (it's a property we should add to `Deno.HttpConn` I think)

Reply:
> (it's a property we should add to Deno.HttpConn I think)

That makes sense, but for std/http users that is no good because the conn is not exposed.

Reply:
Seems more like a WinterCG project to add it to the Request spec?

Reply:
> That makes sense, but for std/http users that is no good because the conn is not exposed.

Things on `Conn` are exposed via the second argument to `serve()` (the `ConnInfo` struct).

Reply:
> > That makes sense, but for std/http users that is no good because the conn is not exposed.
> 
> Things on `Conn` are exposed via the second argument to `serve()` (the `ConnInfo` struct).

I guess that could work.
Still would be nice to be on Request though, especially if the property was also available to set the HTTP version on outgoing requests.